File: java/src/main/java/fr/sdecout/handson/rest/library/LibraryCreationRequest.java
package fr.sdecout.handson.rest.library;

import fr.sdecout.handson.rest.shared.AddressField;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

record LibraryCreationRequest(
        @NotBlank String name,
        @NotNull @Valid AddressField address
) {}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/library/LibraryCreationRequest.kt
package fr.sdecout.handson.rest.library

import fr.sdecout.handson.rest.shared.AddressField
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class LibraryCreationRequest(
    @field:NotBlank val name: String?,
    @field:NotNull @field:Valid val address: AddressField?,
)
